### PR TITLE
action spec adds the execution selector field, and the pipeline selects the executor according to the field

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:

The apitest task will use the apitest executor when it is executed, but version 1.0 should use the schduler executor, version 2.0 only use the apitest executor, so the executor selection field is added to the action spec, and subsequent actions can choose the executor by themselves

action spec add field like: 
```yaml
executor:
  executorKind: "APITEST"
  executorName: "api-test"
```

### CODE REVIEW

File: apistructs/extention.go [hunk 1]
```diff
@@ -240,6 +240,12 @@ type ActionSpec struct {
 	OutputsFromParams []OutputsFromParams   `json:"outputsFromParams" yaml:"outputsFromParams"`
 	Loop              *PipelineTaskLoop     `json:"loop" yaml:"loop"`
 	Priority          *PipelineTaskPriority `json:"priority" yaml:"priority"`
+	Executor          *ActionExecutor       `json:"executor" yaml:"executor"`
+}
+
+type ActionExecutor struct {
+	ExecutorKind string `json:"executorKind" yaml:"executorKind"`
```
Comment (sfwn) [inline apistructs/extention.go:247]: Just `Kind` / `Name` is enough.

File: modules/pipeline/services/pipelinesvc/create.go [hunk 2]
```diff
@@ -189,21 +187,14 @@ func (s *PipelineSvc) makePipelineFromRequest(req *apistructs.PipelineCreateRequ
 	return p, nil
 }
 
-// passedDataWhenCreate stores data passed recursively when create graph.
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/create.go:192]: Put these adjustment into another pr. DO NOT mix it.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Add erda issue link to PR description please.

And in related PRs, you should link this PR.
image

See the effect:
image

**Comment by kakj-go** [comment]
This pr [erda-project(erda-actions#93)](https://github.com/erda-project/erda-actions/pull/93) needs to be merged together

**Comment by kakj-go** [comment]
link issue: [erda-issues](https://terminus-org.app.terminus.io/erda/workBench/projects/387/issues/all?id=73326&issueFilter__urlQuery=eyJzdGF0ZUJlbG9uZ3MiOlsiT1BFTiIsIldPUktJTkciLCJXT05URklYIiwiUkVPUEVOIiwiUkVTT0xWRUQiXSwiYXNzaWduZWVJRHMiOlsiMjAyMCIsIjEwMDA1NjAiXX0%3D&issueTable__urlQuery=eyJwYWdlTm8iOjF9&issueViewGroup__urlQuery=eyJ2YWx1ZSI6ImthbmJhbiIsImNoaWxkcmVuVmFsdWUiOnsia2FuYmFuIjoiZGVhZGxpbmUifX0%3D&iterationID=431&type=BUG)

File: modules/pipeline/services/pipelinesvc/create_task.go [hunk 3]
```diff
@@ -178,11 +179,13 @@ func (s *PipelineSvc) calculateTaskRunAfter(action *pipelineyml.Action) []string
 }
 
 // judgeTaskExecutor judge task executor by action info
-func (s *PipelineSvc) judgeTaskExecutor(action *pipelineyml.Action) (spec.PipelineTaskExecutorKind, spec.PipelineTaskExecutorName, error) {
-	if action.Type == apistructs.ActionTypeAPITest {
-		return spec.PipelineTaskExecutorKindAPITest, spec.PipelineTaskExecutorNameAPITestDefault, nil
+func (s *PipelineSvc) judgeTaskExecutor(action *pipelineyml.Action, actionSpec *apistructs.ActionSpec) (spec.PipelineTaskExecutorKind, spec.PipelineTaskExecutorName, error) {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/create_task.go:182]: Polish code to multi lines.

Comment (sfwn) [reply modules/pipeline/services/pipelinesvc/create_task.go:182]: string = ""

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**